### ISSUE TITLE
Preview URL in comment doesn't match the actual deployment URL

### ISSUE THREAD

**Issue by thomas-noimos** [rossjrw/pr-preview-action#82]
I'm working on a private repo so I hope you not having access to it won't be an issue, but what I'm facing right now is that the comment gives me the following content: 

Deployed preview to https://noimosai.github.io/noimos-ui-kit/pr-preview/pr-18/

However this link doesn't work and gives me a 404. I checked the `gh-pages` branch and the content is there. Apparently it is a new GitHub thing or I configured something wrong but my actual deployment is another URL:

`https://friendly-adventure-lnjz19e.pages.github.io/` and I can access the preview content using the adapted URL `https://friendly-adventure-lnjz19e.pages.github.io/pr-preview/pr-18/`

Here is my config:
![image](https://github.com/rossjrw/pr-preview-action/assets/159126089/254c9479-0bf2-447f-b0bd-133c4aeefb5b)

Did I configure something wrong ?

**Comment by thomas-noimos** [comment]
Sorry I missed the docs where I could configure the `custom-url` prop, closing this.

**Comment by rossjrw** [comment]
Reopening this just in case. While I've not personally seen a `*.pages.github.io` before, it looks like that's what URLs for private enterprise repos look like (based on https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site#about-subdomains-for-privately-published-sites).

`friendly-adventure-lnjz19e` looks randomly-generated - is it permanent? Or does it ever change (over time, with each branch, etc)?

Does `custom-url` completely resolve the issue?

If all looks good, I'll leave this issue open as a documentation task as it'd be good to mention for enterprise users.

**Comment by thomas-noimos** [comment]
@rossjrw yes, using `custom-url` solved the issue. I completely removed the deployment and disabled GH pages, then enabled again and the randomly generated URL was the same after re-enabling it, so I guess it persists.

**Comment by rossjrw** [comment]
Works for me, thanks for the update!

**Comment by CCFenner** [comment]
I have issues when using the action on an enterprise instance.

```
[PR Preview Action](https://<gh-enterprise-host>/rossjrw/pr-preview-action) v1.4.7
:---:
:rocket: Deployed preview to https://<org>.github.io/<repository>/pr-preview/pr-1/
on branch [`gh-pages`](https://<gh-enterprise-host>/<org>/<repository>/tree/gh-pages) at 2024-09-17 07:30 UTC
<!-- Sticky Pull Request Commentpr-preview -->
```

1. The link to this repository actually points to the repo on the enterprise instance, which does not exist.
2. The link to the deployment points to `github.io` instead of `https://pages.<gh-enterprise-host>/<org>/` (I'm using an org site, see [types-of-github-pages-sites](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites))

**Comment by CCFenner** [comment]
> I didn't see any documentation indicating that the Pages for an enterprise instance is definitely hosted at a `pages` subdomain

It's mentioned here https://docs.github.com/en/enterprise-server@3.11/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites
Seems to be related to *subdomain isolation*.

**Comment by rossjrw** [comment]
Aha, thank you! Given that starting point it looks like the only reliable way to detect the Pages URL on Enterprise would be via the Octokit API, which does seem feasible. Unfortunately it doesn't look like I'm going to be able to set up automated testing of this action on Enterprise (https://github.com/orgs/community/discussions/47057).
